### DISCUSSION TITLE
Unable to use the Emgu.CV method CvInvoke.CalibrateCamera()

### DISCUSSION THREAD

**Discussion by tkfc00** [emgucv/emgutf#89] [General]
Dear emgucv user,
I hope , all of you are fine and that i can find a solution for my problem here. For about 5 months I'm programming with csharp using emgu.cv .The paragraph below describe what i did till now.

So now I'm having the image coordinates and the object coordinates and I wish to use the CvInvoke.CalibrateCamera to get out with the projection matrix and the camera matrix. But it is not working since. Please I need help

--------- Below is my code --------------------------------------------------

public static void Test(List finalPoints, Image<Gray,byte> image)
{
// Liste imagepoints
List imagePoints = new List();
List imagesP = new List();

for (int i = 0; i < finalPoints.Count; i++)
{
    imagesP.Add(new PointF() {X = (float)finalPoints[i].OriginalCenter.X ,Y=(float) finalPoints[i].OriginalCenter.Y }); 
}
VectorOfPointF imagePointsvector = new VectorOfPointF();
imagePointsvector.Push(imagesP.ToArray());

for (int i = 0; i < finalPoints.Count; i++)
{
    imagePoints.Add(imagePointsvector);
}

// Konvertieren Sie die List<VectorOfPointF> in ein Array von PointF-Arrays
PointF[][] imagePointsArray = imagePoints.Select(vp => vp.ToArray().Select(p => new PointF((float)p.X, (float)p.Y)).ToArray()).ToArray();

//-------------------

// Liste objectpoints
List<MCvPoint3D32f[]> objectPoints = new List<MCvPoint3D32f[]>();
List<MCvPoint3D32f> objectP = new List<MCvPoint3D32f>();

for (int i = 0; i < finalPoints.Count; i++)
{
    objectP.Add(new MCvPoint3D32f() { X = (float)finalPoints[i].OriginalCenter.X, Y = (float)finalPoints[i].OriginalCenter.Y , 
                                      Z = (float)finalPoints[i].OriginalCenter.Z });
}
MCvPoint3D32f[] objectPointsvector = new MCvPoint3D32f[objectPoints.Count];
Array.Copy(objectP.ToArray() , objectPointsvector , objectPointsvector.Length);

for (int i = 0; i < finalPoints.Count; i++)
{
    objectPoints.Add(objectPointsvector);
}

// Konvertierung die List<MCvPoint3D32f[]> in ein Array von MCvPoint3D32f[][]
MCvPoint3D32f[][] objectPointsArray = objectPoints.Select(vp => vp.ToArray().Select(p => new MCvPoint3D32f((float)p.X, (float)p.Y , 
                                                          (float)p.Z)).ToArray()).ToArray();

// Kalibrierte Kamera-Matrix und Verzerrungskoeffizienten
Mat cameraMatrix = new Mat(3, 3, DepthType.Cv64F, 1);
Mat distortionCoeffs = new Mat(5, 1, DepthType.Cv64F, 1);
Mat[] rotationVectors ;
Mat[] translationVectors ;

CvInvoke.CalibrateCamera(objectPointsArray, imagePointsArray, new Size(image.Width,image.Height), cameraMatrix, 
           distortionCoeffs,CalibType.FixK6, new MCvTermCriteria(1), out rotationVectors, out translationVectors);

}

![T_Testcase2__500px](https://github.com/emgucv/emgutf/assets/126579653/b5620a12-047e-4e5f-ade3-ae820543f5a6)
![T_Testcase5__1000px](https://github.com/emgucv/emgutf/assets/126579653/b8af507e-1b67-4829-a3d8-13a59bf0a509)
![T_Testcase1__500px](https://github.com/emgucv/emgutf/assets/126579653/c86081a3-da97-493d-bf7f-0f2d83c13bc8)

**Comment by emgucv** [comment]
Seems like an Emgu CV (not Emgu TF) discussion. Duplicate to this:
https://github.com/emgucv/emgucv/discussions/899

Closing this dicussion for now.
